DietPi-Software | Beets

nonnorm opened the pull request:
<!--
Before submitting a pull request:
- Please ensure the target branch is "dev" (active development): https://github.com/MichaIng/DietPi/tree/dev
- Please ensure changes have been tested and verified functional.
-->
**Status**: Ready

**Reference**: https://github.com/MichaIng/DietPi/issues/4434

**Commit list/description**:
<!--
- DietPi-Config | Add "Fan control" option to "Performance Options"
![Screenshot](https://xxx.github.com/images/xxx.png)
- DietPi-Config | Add "Fan control" support for Odroid C2
- DietPi-Config | Syntax fix
-->
- DietPi-Software | Add Beets installation option

nonnorm committed:
v7.3
+ DietPi-Software | Add Beets installation option

nonnorm committed:
v7.3
+ DietPi-Software | Add Beets installation option

nonnorm committed:
v7.3
+ DietPi-Software | Fix capitalization

nonnorm committed:
v7.3
+ DietPi-Software | Fix capitalization

nonnorm committed:
v7.3
+ DietPi-Software | Remove bashrc.d file

nonnorm committed:
v7.3
+ DietPi-Software | Remove bashrc.d file

nonnorm committed:
v7.3
+ DietPI-Software | Source beets bashrc.d file

nonnorm committed:
v7.3
+ DietPI-Software | Source beets bashrc.d file

nonnorm committed:
v7.3
+ DietPi-Software | Fix G_EXEC typo

nonnorm committed:
v7.3
+ DietPi-Software | Fix G_EXEC typo

nonnorm committed:
v7.3
+ DietPi-Software | Add shebang to beets bashrc.d script

nonnorm committed:
v7.3
+ DietPi-Software | Add shebang to beets bashrc.d script

nonnorm committed:
Merge branch 'beets' of https://github.com/MichaIng/DietPi into beets

MichaIng committed:
v7.3 (#4441)

META | Update beets branch

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -652,6 +652,14 @@ INDEX_BROWSER_TARGET=$INDEX_BROWSER_TARGET"
+		aSOFTWARE_DESC[$software_id]='music organizer and manager'
+		aSOFTWARE_CATX[$software_id]=2
+		aSOFTWARE_DOCS[$software_id]='https://dietpi.com/docs/software/media/#beets'
+		aSOFTWARE_DEPS[$software_id]='130'
```
130 is actually Python pip, while here only the runtime is required. As we install the APT package which has all required Python modules as dependency, I think we can and should remove the dependency here.

It is btw possible to install it via `pip3 install beets`, which brings a newer version and then requires 130. Not sure whether it's worth it, as it needs to be compiled, hence requires the whole build-essentials stack as well.

nonnorm replied:
The Debian package is only 2 minor versions behind, so I think that we'll be fine using that.

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -2971,6 +2979,21 @@ _EOF_
+
+			Banner_Installing
+
+			G_EXEC mkdir /etc/beets
```
As this is not only used for the configuration, but also for the database, what do you think about using `/mnt/dietpi_userdata/beets` to align this with other cases?

FHS would be `/var/lib/beets/library.db`, but IMO it's nice to have everything in userdata, as long as it's not somehow preset by the APT package.

nonnorm replied:
The preset is actually `~/.config/beets`, but the `bashrc.d` script overrides that.

MichaIng replied:
Jep, I thought that and makes sense when we expect a single music library on the system 👍.

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -2971,6 +2979,21 @@ _EOF_
+			G_EXEC mkdir /etc/beets
+			G_EXEC chmod 777 /etc/beets
+			
+			G_EXEC eval "echo -e '#!/bin/dash\nexport BEETSDIR=/etc/beets' > /etc/bashrc.d/dietpi-beets.sh"
```
No need for the shebang. Using dash is even confusing as `/etc/bashrc.d` of course is only loaded for interactive bash shells. As it is sourced, not executed, no shebang is required. Not sure if we use a shebang in other such cases, but then it should be removed.

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -8215,6 +8238,18 @@ _EOF_
+
+			Banner_Configuration
+
+			cat << _EOF_ > /etc/beets/config.yaml
```
On reinstall (if the file exists already), we should not overwrite it.

nonnorm committed:
v7.3
+ DietPi-Software | Fixes

MichaIng committed:
v7.3

+ DietPi-Software | Beets: Merge install and config blocks, as both as very small and we anyway want to merge them for all software titles soon
+ DietPi-Software | Beets: Apply switch to /mnt/dietpi_userdata/beets for all steps
+ DietPi-Software | Beets: Limit write access to library to dietpi user and members of audio group

MichaIng committed:
v7.3

+ Typo

MichaIng committed:
v7.3

+ DietPi-Software | Beets: Pre-create state file as well and apply group write permissions to whole directory. Somehow managing the library does not work when beets dir write permissions are granted, despite the fact that no file needs to be created or removed (?).

MichaIng committed:
v7.3

+ README | Add Beets to 3rd party sources list

MichaIng committed:
Merge branch 'dev' into beets

MichaIng committed:
v7.3

+ CHANGELOG | Beets: This command line music organizer and manager has been added with software ID 190.

MichaIng committed:
Merge branch 'dev' into beets
